web/gui/dashboard/src/dashboard.js

```diff
@@ -0,0 +1,9951 @@
+/*global netdataAlarmsNotifCallback  *//* function, a hook for alarm notifications
+ *                                                  (default: undefined) */
+/*global netdataIntersectionObserver *//* boolean,  enable or disable the use of intersection observer
+ *                                                  (default: true) */
+/*global netdataCheckXSS             *//* boolean,  enable or disable checking for XSS issues
+ *                                                  (default: false) */
+
+/* global NETDATA */
+/* eslint-disable */
+
+// *** src/dashboard.js/utils.js
+
+import { updateChartDataAction, updateChartDetailsAction } from './domains/chart/actions';
+
+let reduxStore
+NETDATA.name2id = function (s) {
+    return s
+        .replace(/ /g, '_')
+        .replace(/:/g, '_')
+        .replace(/\(/g, '_')
+        .replace(/\)/g, '_')
+        .replace(/\./g, '_')
+        .replace(/\//g, '_');
+};
+
+NETDATA.encodeURIComponent = function (s) {
+    if (typeof(s) === 'string') {
+        return encodeURIComponent(s);
+    }
+
+    return s;
+};
+
+NETDATA.guid = function () {
+    function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+            .toString(16)
+            .substring(1);
+    }
+
+    return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
+};
+
+NETDATA.zeropad = function (x) {
+    if (x > -10 && x < 10) {
+        return '0' + x.toString();
+    } else {
+        return x.toString();
+    }
+};
+
+NETDATA.seconds4human = function (seconds, options) {
+    let defaultOptions = {
+        now: 'now',
+        space: ' ',
+        negative_suffix: 'ago',
+        day: 'day',
+        days: 'days',
+        hour: 'hour',
+        hours: 'hours',
+        minute: 'min',
+        minutes: 'mins',
+        second: 'sec',
+        seconds: 'secs',
+        and: 'and'
+    };
+
+    if (typeof options !== 'object') {
+        options = defaultOptions;
+    } else {
+        for (var x in defaultOptions) {
+            if (typeof options[x] !== 'string') {
+                options[x] = defaultOptions[x];
+            }
+        }
+    }
+
+    if (typeof seconds === 'string') {
+        seconds = parseInt(seconds, 10);
+    }
+
+    if (seconds === 0) {
+        return options.now;
+    }
+
+    let suffix = '';
+    if (seconds < 0) {
+        seconds = -seconds;
+        if (options.negative_suffix !== '') {
+            suffix = options.space + options.negative_suffix;
+        }
+    }
+
+    let days = Math.floor(seconds / 86400);
+    seconds -= (days * 86400);
+
+    let hours = Math.floor(seconds / 3600);
+    seconds -= (hours * 3600);
+
+    let minutes = Math.floor(seconds / 60);
+    seconds -= (minutes * 60);
+
+    let strings = [];
+
+    if (days > 1) {
+        strings.push(days.toString() + options.space + options.days);
+    } else if (days === 1) {
+        strings.push(days.toString() + options.space + options.day);
+    }
+
```

Comment:
I will write a doubt here that will be the same for the next blocks of "if" and "else if", you are working with values for "hours", "minutes" and "second" when they are bigger than 1 and equal 1, but these three variables also accepts the value 0. Why are you not working with zeros?

Reply:
the zeros are not taken into account because when there are 0 days/hours/minutes/seconds we simply don't append anything to the string. for example `seconds4human(60 * 60) ==="1 hour"`

Reply:
I saw that we were not appending nothing to string, this was something that let me worried, because we will have in different times something like: "2019-08-07 00:00:00" or "2019-08-07 01:00:00" and so on.

About the day is not necessary to worry, the day 0 is far from independent of the most common 0 reference that we will get to test.